### DISCUSSION TITLE
Body parsing is failing when calling the API from any browser

### DISCUSSION THREAD

**Discussion by ryshoooo** [fastapi/fastapi#6692] [Questions] [answered]
### TL;DR
When calling a FastAPI endpoint which expects a JSON body from a frontend framework, I receive a response of 422 - Unprocessable entity, even though the request is correctly formatted.

### Example

_requirements.txt_
```bash
fastapi==0.65.2
uvicorn==0.14.0
```

_app.py_
```Python
from fastapi import FastAPI
from pydantic import BaseModel
from typing import Optional, List


class SimpleInput(BaseModel):
    NRows: Optional[int] = 0
    Cols: Optional[List[str]] = []


app = FastAPI()


@app.post("/test/{param1}")
def test_post(param1: str, inputs: SimpleInput):
    return {"Param1": param1, "Inputs": inputs}
```

### Steps to reproduce

* Start a virtual environment `python -m venv venv` and activate it `source venv/bin/activate`
* Install all the dependencies `pip install -r requirements.txt`
* Start the application `uvicorn app:app --reload --host 0.0.0.0 --port 8000`
* Go to browser `http://localhost:8000/docs`
* Open browser console and execute `fetch("http://localhost:8000/test/saf", {method: "POST", body: JSON.stringify({NRows: 10, Cols: []})})`

You should get a response of 422. Interestingly, executing the call from the docs pages by clicking around does not resolve in 422. Also, this does not happen in FastAPI 0.65.1, only in 0.65.2.

### Environment

* OS: Linux, macOS
* FastAPI Version: 0.65.2
* Python version: 3.7, 3.8, 3.9, 3.10-rc

### Additional context
If this was a change done on purpose (i.e. a feature), I'd highly advise reconsidering this decision. Happens in Chrome, Safari, Firefox (I assume the browser is not the issue, so I'll just generalize and assume that it happens in any browser).

**Comment by Kludex** [comment, marked as answer]
You need to add the `Content-Type` header with `application/json` value. 

This behavior is expected as per https://github.com/tiangolo/fastapi/releases/tag/0.65.2

**Comment by ryshoooo** [comment]
> You need to add the `Content-Type` header with `application/json` value.
> 
> This behavior is expected as per https://github.com/tiangolo/fastapi/releases/tag/0.65.2

That makes a lot of sense :) and fixed it. Thanks!
